src/Http/Http/src/Features/QueryFeature.cs

```diff
@@ -100,5 +102,237 @@ public IQueryCollection Query
                 }
             }
         }
+
+        /// <summary>
+        /// Parse a query string into its component key and value parts.
+        /// </summary>
+        /// <param name="queryString">The raw query string value, with or without the leading '?'.</param>
+        /// <returns>A collection of parsed keys and values, null if there are no entries.</returns>
+        [SkipLocalsInit]
+        internal static AdaptiveCapacityDictionary<string, StringValues>? ParseNullableQueryInternal(string? queryString)
```

Comment:
QueryHelpers had more test coverage that you'll need to port over:
https://github.com/dotnet/aspnetcore/blob/main/src/Http/WebUtilities/test/QueryHelpersTests.cs

Reply:
--> a6353d151439e1ab7aa08e1a02fa660e6d1dcff5